cmd/nodecmd/setup.go

```diff
@@ -0,0 +1,231 @@
+	cmd.Flags().BoolVar(&publicHTTPPortAccess, "public-http-port", false, "allow public access to avalanchego HTTP port")
+	cmd.Flags().StringArrayVar(&nodeIPs, "node-ips", []string{}, "IP addresses of nodes")
+	cmd.Flags().StringArrayVar(&sshKeyPaths, "ssh-key-paths", []string{}, "ssh key paths")
+	cmd.Flags().BoolVar(&useSSHAgent, "use-ssh-agent", false, "use ssh agent(ex: Yubikey) for ssh auth")
+	cmd.Flags().StringVar(&genesisPath, "genesis", "", "path to genesis file")
+	cmd.Flags().StringVar(&upgradePath, "upgrade", "", "path to upgrade file")
+	cmd.Flags().BoolVar(&partialSync, "partial-sync", true, "primary network partial sync")
+	return cmd
+}
+
+func setup(hosts []*models.Host, avalancheGoVersion string, network models.Network) error {
+	if globalNetworkFlags.UseDevnet {
+		partialSync = false
+		ux.Logger.PrintToUser("disabling partial sync default for devnet")
+	}
+	ux.Logger.PrintToUser("Starting bootstrap process on the newly created Avalanche node(s)...")
+	wg := sync.WaitGroup{}
+	wgResults := models.NodeResults{}
+	spinSession := ux.NewUserSpinner()
+
+	for _, host := range hosts {
+		wg.Add(1)
+		go func(nodeResults *models.NodeResults, host *models.Host) {
+			defer wg.Done()
+			if err := host.Connect(0); err != nil {
+				nodeResults.AddResult(host.IP, nil, err)
+				return
+			}
+			if err := provideStakingCertAndKey(host); err != nil {
+				nodeResults.AddResult(host.IP, nil, err)
+				return
+			}
+			spinner := spinSession.SpinToUser(utils.ScriptLog(host.IP, "Setup Node"))
+			if err := ssh.RunSSHSetupNode(host, app.Conf.GetConfigPath()); err != nil {
+				nodeResults.AddResult(host.IP, nil, err)
+				ux.SpinFailWithError(spinner, "", err)
+				return
+			}
+			if err := ssh.RunSSHSetupDockerService(host); err != nil {
+				nodeResults.AddResult(host.IP, nil, err)
+				ux.SpinFailWithError(spinner, "", err)
+				return
+			}
+			ux.SpinComplete(spinner)
+			spinner = spinSession.SpinToUser(utils.ScriptLog(host.IP, "Setup AvalancheGo"))
+			// check if host is a API host
+			if err := docker.ComposeSSHSetupNode(host,
+				network,
+				avalancheGoVersion,
+				bootstrapIDs,
+				bootstrapIPs,
+				partialSync,
+				genesisPath,
+				upgradePath,
+				addMonitoring,
+				host.APINode); err != nil {
+				nodeResults.AddResult(host.IP, nil, err)
+				ux.SpinFailWithError(spinner, "", err)
+				return
+			}
+			ux.SpinComplete(spinner)
+		}(&wgResults, host)
+	}
+	wg.Wait()
+	spinSession.Stop()
+	for _, node := range hosts {
+		if wgResults.HasIDWithError(node.NodeID) {
+			ux.Logger.RedXToUser("Node %s has ERROR: %s", node.IP, wgResults.GetErrorHostMap()[node.IP])
+		}
+	}
+
+	if wgResults.HasErrors() {
+		return fmt.Errorf("failed to deploy node(s) %s", wgResults.GetErrorHostMap())
+	} else {
+		ux.Logger.PrintToUser(logging.Green.Wrap("AvalancheGo and Avalanche-CLI installed and node(s) are bootstrapping!"))
+	}
+	return nil
+}
+
+func setupNode(_ *cobra.Command, _ []string) error {
```

Comment:
we need to verify that provided hosts are Ubuntu OS and we have enough access to do what we need

Reply:
another suggestion is to verify that host meets minimum requirements( for aws provision we control this, but for user provided IP we don't know what the box is)